### PR TITLE
Fully Implement Game Logic

### PR DESCRIPTION
**Pull request by matyimat**
Without testing

### CODE REVIEW

File: Dixit_Logic/Classes/DixitGame.cs [hunk 1]
```diff
@@ -49,10 +49,15 @@ public IGameState ActualGameState
         }
 
         /// <summary>     
-        /// This event is triggered at the start of a trun when all cards are consumed from the MainDeck and players have no more cards too .
+        /// This event is triggered when all cards are consumed from the MainDeck and players have no more cards too .
```
Comment (VSZM) [inline Dixit_Logic/Classes/DixitGame.cs]: If this comment is true, then the implementation is wrong. A Dixit game lasts until one of the players reach 30 points. Playing until all cards are gone is not a reliable metric as the number of cards can be vast.

File: Dixit_Logic/Classes/GameState.cs [hunk 2]
```diff
@@ -71,15 +71,27 @@ public class GameState : IGameState
...
+            _hands = new Dictionary<IPlayer, IDeck>();        
             _players = new List<IPlayer>();            
             _points = new Dictionary<IPlayer, int>();
             _roundStatus = PhaseStatus.BeforeStart; // set to the game's round starting status
+
+            
+            List<ICard> baseCardList = new List<ICard>();
+
+            const int numberOfCards = 84;
+            //add cards to main deck
```
Comment (VSZM) [inline Dixit_Logic/Classes/GameState.cs:100]: @matyimat How do you know the number of cards? This should not be hardcoded, as this part of the code is not responsible for the data. Data layer provides an interface for getting back the list of cards.

**Review by VSZM — CHANGES_REQUESTED** [review]
Please fix the requested changes

File: Dixit_Logic/Classes/DixitGame.cs [hunk 3]
```diff
@@ -133,11 +161,116 @@ public bool StartGame()
...
         {
-            throw new NotImplementedException();
+            if (_actGameState.RoundStatus == PhaseStatus.RoundEnd)
+            {
+                Card originalCard = null;
+                foreach (var boardCard in _actGameState.BoardDeck.Cards)
+                {
+                    if (((Card)boardCard).Owner.Equals(_actGameState.ActualPlayer))
+                    {
+                        originalCard = (Card)boardCard;
```
Comment (SilverGeri) [inline Dixit_Logic/Classes/DixitGame.cs:177]: you can add a `break` here

File: Dixit_Logic/Classes/DixitGame.cs [hunk 4]
```diff
@@ -133,11 +161,116 @@ public bool StartGame()
...
+
+                    if (actPlayerIndex >= (_actGameState.Players.Count - 1))
+                    {
+                        actPlayerIndex = 0;
+                    }
+                    else
+                    {
+                        actPlayerIndex++;
+                    }
+
```
Comment (SilverGeri) [inline Dixit_Logic/Classes/DixitGame.cs:266]: you could just do `actPlayerIndex = (actPlayerIndex +1)%Players.Count`

File: Dixit_Logic/Classes/DixitGame.cs [hunk 5]
```diff
@@ -133,11 +161,116 @@ public bool StartGame()
...
+                    _actGameState.ActualPlayer = _actGameState.Players.ElementAt(actPlayerIndex);
+                    _actGameState.RoundStatus = PhaseStatus.AssociationTelling;
+
+                    //if it prepares the new round than it raises the new round event
+                    NewRound(this, new EventArgs());
```
Comment (SilverGeri) [inline Dixit_Logic/Classes/DixitGame.cs:271]: @selleim please don't forget to handle this event. (I know this comment doesn't really belong here, but the deadline is near)

File: Dixit_Logic/Interfaces/IGameState.cs [hunk 6]
```diff
@@ -91,5 +91,14 @@ IDeck BoardDeck
         {
             get;
         }
+
+        /// <summary>
+        /// Dixit game lasts until one of the players reach this point.
+        /// </summary>
+        int ObtainablePoint
```
Comment (SilverGeri) [inline Dixit_Logic/Interfaces/IGameState.cs:98]: where can you reset it? I saw you hardcoded 5 in GameState.cs but if we want to play with another obtainable point, do we have to rebuild the project? I think this property should be public

Comment (matyimat) [reply Dixit_Logic/Interfaces/IGameState.cs:98]: This is public, i cant add acces modifier here, only in the class what implement the interface (GameState)

**Review by VSZM — APPROVED** [review]

**Pull request merged**